nisystemlink/clients/alarm/models/_query_alarms_request.py

```diff
@@ -0,0 +1,93 @@
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional, Union
+
+from nisystemlink.clients.core._uplink._json_model import JsonModel
+
+
+class TransitionInclusionOption(str, Enum):
+    """Determines which transitions to include in the query results."""
+
+    NONE = "NONE"
+    """No transitions are included."""
+
+    MOST_RECENT_ONLY = "MOST_RECENT_ONLY"
+    """The most recent transition is included."""
+
+    ALL = "ALL"
+    """All transitions are included."""
+
+
+class AlarmOrderBy(str, Enum):
+    """The sort order of the returned list of alarms."""
+
+    UPDATED_AT = "UPDATED_AT"
+    """The date and time the alarm was last updated."""
+
+
+class QueryAlarmsWithFilterRequest(JsonModel):
+    """Contains filter information for querying alarms."""
+
+    filter: Optional[str] = None
+    """The alarm query filter in Dynamic Linq.
+
+    Allowed properties in the filter are: acknowledged, acknowledgedAt, active, alarmId,
```

Comment:
Consider an easier-to-read list such as what we have in the [DataFrame Service client](https://github.com/ni/nisystemlink-clients-python/blob/6b28ef60f4c0b46f83d33f73b5fd0986c945a779/nisystemlink/clients/dataframe/models/_query_tables_request.py#L13) and what the Alarm Service's OpenAPI document uses:

<img width="1109" height="925" alt="Image" src="https://github.com/user-attachments/assets/2ab8dc4e-47bc-460f-a8e7-544c950e1781" />